include/boost/geometry/strategies/spherical/distance_cross_track_point_box.hpp

```diff
@@ -32,131 +42,223 @@ namespace boost { namespace geometry
+        }
+        else
+        {
+            // the box does not cross the antimeridian
 
-                return_type const projection1 = cos( d_crs1 );
-                return_type const projection2 = cos( d_crs2 );
+            return_type const lon_sum = lon_min + lon_max;
+            if (math::equals(lon_sum, return_type(0)))
+            {
+                // special case: the box is symmetric with respect to
+                // the prime meridian; the midway meridian is the antimeridian
 
-                if(projection1 > 0.0 && projection2 > 0.0)
+                if (plon < lon_min)
                 {
-                    return_type const d1 = m_pp_strategy.apply(p1, point);
-                    return_type const
-                        XTD = radius()
-                            * geometry::math::abs(
-                                asin( sin( d1 / radius() ) * sin_d_crs1 )
-                              );
-
-                    return return_type(XTD);
+                    return m_ps_strategy.apply(point, bottom_left, top_left);
                 }
                 else
                 {
-                    // OPTIMIZATION
-                    // Return d1 if projection1 <= 0 and d2 if projection2 <= 0
-                    // if both == 0 then return d1 or d2
-                    // both shouldn't be < 0
+                    return m_ps_strategy.apply(point, bottom_right, top_right);
+                }
+            }
 
-                    return_type const d1 = m_pp_strategy.apply(p1, point);
-                    return_type const d2 = m_pp_strategy.apply(p2, point);
+            // midway longitude = lon_min - (2 * pi - (lon_max - lon_min)) / 2;
+            return_type lon_midway = (lon_min + lon_max) / two - pi;
 
-                    return return_type((std::min)( d1 , d2 ));
+            // normalize the midway longitude
+            if (lon_midway > pi)
+            {
+                lon_midway -= two_pi;
+            }
+            else if (lon_midway < -pi)
+            {
+                lon_midway += two_pi;
+            }
+            BOOST_GEOMETRY_ASSERT(lon_midway >= -pi && lon_midway <= pi);
+
+            if (lon_sum > 0)
+            {
+                // the midway meridian is left of the box
+                if (plon < lon_min && plon >= lon_midway)
+                {
+                    return m_ps_strategy.apply(point, bottom_left, top_left);
+                }
+                else
+                {
+                    return m_ps_strategy.apply(point, bottom_right, top_right);
+                }
+            }
+            else
+            {
+                // the midway meridian is right of the box
+                if (plon > lon_max && plon <= lon_midway)
+                {
+                    return m_ps_strategy.apply(point, bottom_right, top_right);
+                }
+                else
+                {
+                    return m_ps_strategy.apply(point, bottom_left, top_left);
```

Comment:
Do I understand correctly that those conditions requires that the plon must be in the valid [-pi, pi] range?

I also have a small suggestion. I'm guessing that if some boolean value indicating which corner should be used was calculated at the beginning, and then after all of those conditions and setting this boolean value a strategy was called with corresponding parameters, it'd allow the compiler to optimize the code more easily. Now in each branch there is a potentially inlineable function call which could result in a code bloat and worse instruction cachng. With this boolean only this variable, most certainly optimized out would be calculated and then there would be only one if/else with function calls. So:

    if (...) {
        temp = ...
        if (...)
            use_left = true;
        else
            use_left = false;
    } else {
        temp = ...
        if (...)
            use_left = false;
        else
            use_left = true;
    }
    if ( use_left )
        m_ps_strategy.apply(point, bottom_left, top_left);
    else
        m_ps_strategy.apply(point, bottom_right, top_right);

This theory could of course be verified but still, if those strategies was called in one place, at the end, the code would probably also be more clear and safe since there would be less function calls into which wrong parameters could be passed.

Not to mention that it would probably be possible to calculate this boolean variable using terniary `?:` operators or at least use it for the calculation of partial results. Or maybe even to write everything in one if condition, though this would probably make the code less readable.